File: Java/Easy/WrongAnswer/shortest-distance-to-target-string-in-a-circular-array_7.java
class Solution {
    public int closetTarget(String[] words, String target, int startIndex) {
        
        int minDiff = Integer.MAX_VALUE;
        int i = startIndex;
        int j = startIndex;
        
        while (i >= 0 || j < words.length) {
            if (i >= 0 && words[i].equals(target)) {
                minDiff = Math.min(minDiff, startIndex - i);
                break;
            }
            
            if (j < words.length && words[j].equals(target)) {
                minDiff = Math.min(minDiff, j - startIndex);
                break;
            }
            
            i--;
            j++;
        }
        
        return minDiff == Integer.MAX_VALUE ? -1 : minDiff;
    }
}

File: Rust/Easy/WrongAnswer/shortest-distance-to-target-string-in-a-circular-array_7.rs
impl Solution {
    pub fn closet_target(words: Vec<String>, target: String, start_index: i32) -> i32 {
        
        let mut n = words.len();
        let mut i = start_index as usize;
        let mut j = start_index as usize;
        let mut k = 0;
        let mut res = n as i32;
        while k < n {
            if words[i] == target {
                res = res.min((i as i32 - start_index).abs());
            }
            if words[j] == target {
                res = res.min((j as i32 - start_index).abs());
            }
            i = (i + 1) % n;
            j = (j + n - 1) % n;
            k += 1;
        }
        res
    }
}